### PR TITLE
feat(NFDIV-440): Added Save & sign out link with session storage update

### PR DESCRIPTION
**Pull request by SebFlippence**
### JIRA link ###

Part of [NFDIV-440](https://tools.hmcts.net/jira/browse/NFDIV-440) English Version - Who are you applying to divorce / end a civil partnership with

### Change description ###

* Adds the "Save and sign out" link at the bottom of the form, which triggers a form submit, saving the current state without validation
* Sends the current form data session state to the current step/page
* Fixes an issue where the session state was not always saving
* Passes the current form state back into the form to restore the users input

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/controller/GetController.ts [hunk 1]
```diff
@@ -10,7 +10,11 @@ export type TranslationFn = (isDivorce: boolean) => Translations;
 
 @autobind
 export class GetController {
-  constructor(protected readonly view: string, protected readonly content: TranslationFn | Translations) {}
+  constructor(
+    protected readonly view: string,
+    protected readonly content: TranslationFn | Translations,
+    protected readonly id: string = 'page'
```
Comment (velimd) [inline src/main/app/controller/GetController.ts:16]: how comes you are setting a default value here?

File: src/main/modules/nunjucks/index.ts [hunk 2]
```diff
@@ -19,11 +19,11 @@ export class Nunjucks {
       return typeof prop === 'function' ? prop(this.ctx) : prop;
     });
 
-    env.addGlobal('formItems', function (items: FormInput[]) {
+    env.addGlobal('formItems', function (items: FormInput[], selected: string) {
       return items.map(i => ({
         text: this.env.globals.getContent.call(this, i.label),
         value: i.value,
-        selected: i.selected,
+        checked: i.selected || i.value === selected,
```
Comment (velimd) [inline src/main/modules/nunjucks/index.ts:26]: im confused on i.selected, would a form ever have a default value other than the one the user selected before and saved in session?

Comment (velimd) [reply src/main/modules/nunjucks/index.ts:26]: i guess its to be generic component.

File: src/main/app/controller/GetController.ts (hunk 1 above)
Comment (SebFlippence) [reply src/main/app/controller/GetController.ts]: 👍 Good spot, I've renamed this to `stepId` to make it more obvious what it's referring to and set it to undefined by default for pages that don't need a form in bb0bf46.

File: src/main/modules/nunjucks/index.ts (hunk 2 above)
Comment (SebFlippence) [reply src/main/modules/nunjucks/index.ts]: Yeah, this is for generic input options. `i.selected` is from the `form` object in `content.ts`. So you could always make sure that an option is always selected by default. The second part is the users selection so I've renamed this to be `userSelection` in b07abed 👍

**Review by velimd — APPROVED** [review]

**Pull request merged**